qemu/tests/qnum_change_reboot.py

```diff
@@ -0,0 +1,209 @@
+            n_client.session = session_serial
+            error_context.context("Enable multi queues in guest.", logging.info)
+            for nic in vm.virtnet:
+                ifname = utils_net.get_linux_ifname(session_serial, nic.mac)
+                queues = int(nic.queues)
+                change_queues_number(ifname, queues)
+            error_context.context("Run test %s background" % netperf_stress,
+                                  logging.info)
+            stress_thread = utils_misc.InterruptedThread(netperf_test)
+            stress_thread.start()
+            utils_misc.wait_for(lambda: wait_time, 0, 5,
+                                "Wait %s start background" % netperf_stress)
+
+            # ping test
+            error_context.context("Ping guest from host", logging.info)
+            args = (guest_ip, ping_time, ping_lost_ratio)
+            bg_ping = utils_misc.InterruptedThread(ping_test, args)
+            bg_ping.start()
+
+            error_context.context("Change queues number repeatedly",
+                                  logging.info)
+            repeat_counts = params.get_numeric("repeat_counts")
+            for nic in vm.virtnet:
+                queues = int(nic.queues)
+                if queues == 1:
+                    logging.info("Nic with single queue, skip and continue")
+                    continue
+                ifname = utils_net.get_linux_ifname(session_serial, nic.mac)
+                change_list = params.get("change_list").split(",")
+                for repeat_num in range(repeat_counts):
+                    error_context.context("Change queues number -- %sth"
+                                          % repeat_num, logging.info)
+                    queues_status = get_queues_status(ifname)
+                    for q_number in change_list:
+                        queues_status = change_queues_number(ifname,
+                                                             int(q_number),
+                                                             queues_status)
+
+            logging.info("wait for background test finish")
+            try:
+                stress_thread.join()
+            except Exception as err:
+                err_msg = "Run %s test background error!\n "
+                err_msg += "Error Info: '%s'"
+                test.error(err_msg % (netperf_stress, err))
+
+            logging.info("Wait for background ping test finish.")
+            try:
+                bg_ping.join()
+            except Exception as err:
+                txt = "Fail to wait background ping test finish. "
+                txt += "Got error message %s" % err
+                test.fail(txt)
+
+            if required_reboot and session_serial:
+                logging.info("Rebooting guest ...")
+                vm.reboot()
+                required_reboot = False
+            else:
+                bg_test = False
+    finally:
```

Comment:
You may need invoke cleanup() for n_server and n_client.

Reply:
I will add test parameters when update next time.